src/backend/marsha/core/tests/test_api_xapi_statement.py

```diff
@@ -234,6 +242,8 @@ def test_xapi_statement_with_missing_user_id(self, video_model_mock):
         }
 
         video_model_mock.return_value = video
+        xapi_instance = xapi_mock.return_value
+        xapi_instance.send.return_value = None
```

Comment:
I don't get these 2 lines can you explain?

Reply:
I mock the send method returning `None`

Reply:
I replaced it with 

```
        with mock.patch("marsha.core.api.XAPI.send", return_value=None):
            response = self.client.post(
                "/xapi/",
                HTTP_AUTHORIZATION="Bearer {!s}".format(jwt_token),
                data=json.dumps(data),
                content_type="application/json",
            )
```

Reply:
But if I'm not mistaken, mocking the `send` method is the only reason for `xapi_mock`?
So you could replace the mock in decorator by:
```python
from marsha.core.api import XAPI
...
@mock.patch.object(XAPI, "send", return_value=None)
```

Reply:
Yes but using a context manager is not better here ? I only want to patch, I don't use the mock.

Reply:
yes you can use a context manager but then use what I proposed for decorator as a context manager and you can remove the decorator.

Reply:
It's what I did :+1: